rewrite_models.py

```diff
@@ -47,15 +49,24 @@ def rewrite_model(filename, inputs_dict, module_dict, attr_dict, forward_list):
             f.write(opstr.replace(",)\n", ")\n"))
         f.write("\n")
         f.write("m = M().eval()\n")
-        # f.write("ref_m = "+pretrained_models_str[modelname]+".eval()\n")
+        f.write("CORES=os.popen(\"lscpu | grep Core | awk '{print $4}'\").readlines()\n")
+        f.write("SOCKETS=os.popen(\"lscpu | grep Socket | awk '{print $2}'\").readlines()\n")
+        f.write("BS=int(CORES[0])*int(SOCKETS[0])\n")
+        f.write("batch_size=BS\n")
         for k in inputs_dict:
             f.write(k+" = " + inputs_dict[k] + "\n")
-        # f.write("ref_output = ref_m("+inputstr+")\n")
-        f.write("start = time.time()\n")
-        f.write("output = m("+inputstr+")\n")
-        f.write("end = time.time()\n")
-        f.write("print(end-start)\n")
-        # f.write("print(ref_output[0].shape==output[0].shape)\n")
+        f.write("def print_throughput(flag):\n")
+        f.write("    start_time=time.time()\n")
+        f.write("    for i in range(10):\n")
+        f.write("        output = m("+inputstr+")\n")
+        f.write("    total_iter_time = time.time() - start_time\n")
+        f.write("    Throughput = batch_size * 10 / total_iter_time\n")
+        f.write("    file_current = os.path.basename(__file__)\n")
+        f.write("    print(file_current,',',BS,',',flag,',',Throughput)\n")
+
+        f.write("for flag in {False,True}:\n")
+        f.write("    torch._C._jit_set_texpr_fuser_enabled(flag)\n")
+        f.write("    print_throughput(flag)\n")
```

Comment:
This benchmark is not only used for NNC. I think the generated models should be general so that we do not need to spend time analyzing models and generating codes every time we have new requirements. So such specific settings need to be decoupled from auto-gen scripts. What do you think about it?

Reply:
ok